asset/asset.go

```diff
@@ -798,39 +820,57 @@ func NewScriptKeyBip86(rawKey keychain.KeyDescriptor) ScriptKey {
 	}
 }
 
-// DeriveGroupKey derives an asset's group key based on an internal public
-// key descriptor, the original group asset genesis, and the asset's genesis.
-func DeriveGroupKey(genSigner GenesisSigner, genBuilder GenesisTxBuilder,
-	rawKey keychain.KeyDescriptor, initialGen Genesis,
-	newAsset *Asset) (*GroupKey, error) {
+// ValidateGroupKeyRequest ensures that the asset intended to be a member of an
+// asset group is well-formed.
+func (req *GroupKeyRequest) ValidateGroupKeyRequest() error {
```

Comment:
We could create a `NewGroupKeyRequest` function and add these validation checks within that function. That way we would avoid the risk of forgetting to call `ValidateGroupKeyRequest`. Or we could do both because the validation is cheap.

Either way, we should probably name this method `Validate`.

Reply:
Updated the name; fair point re: having this happen in some constructor instead of a separate func as it is now, I guess the `TapscriptRoot` is the only optional field.